### PR TITLE
✅ Tests for signup 100%

### PR DESCRIPTION
**Pull request by Wolfy64**
- Add [React Testing Library](https://testing-library.com/docs/react-testing-library/intro) 🐐
- Fix CSS focus for each error field
- Add tests to cover signup up to 100%
- Update Typescript on `<Input />` and his stories to add tests

### CODE REVIEW

File: __tests__/pages/signup.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  test('Should not submit values', async () => {
+    const { getByTestId } = render(<Signup {...props} />)
+    const submitButton = getByTestId('submit')
+    fireEvent.click(submitButton)
+    await wait(() => expect(submitSignup).not.toBeCalled())
+  })
+
+  test('Should submit values', async () => {
+    const { getByTestId } = render(<Signup {...props} />)
+    const emailField = getByTestId('email')
```
Comment (bryanjenningz) [inline __tests__/pages/signup.js]: Consider using getByPlaceholderText: https://testing-library.com/docs/dom-testing-library/api-queries#byplaceholdertext

File: __tests__/pages/signup.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    const usernameField = getByTestId('username')
+    const passwordField = getByTestId('password')
+    const firstNameField = getByTestId('firstName')
+    const lastNameField = getByTestId('lastName')
+    const submitButton = getByTestId('submit')
+
+    await wait(() =>
```
Comment (hwong0305) [inline __tests__/pages/signup.js]: Can you please consolidate all the `fireEvent.change` into a single `wait()`?

File: __tests__/pages/signup.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+      fireEvent.change(lastNameField, {
+        target: {
+          value: 'name'
+        }
+      })
+    )
+
+    fireEvent.click(submitButton)
+
+    await wait(() => expect(submitSignup).toHaveBeenCalledTimes(1))
```
Comment (hwong0305) [inline __tests__/pages/signup.js]: Please consolidate with the `wait()` below.

File: __tests__/pages/signup.js (hunk 2 above)
Comment (Wolfy64) [reply __tests__/pages/signup.js]: @hwong0305 Done!

**Pull request merged**